geode-core/src/main/java/org/apache/geode/cache/query/internal/index/DummyQRegion.java

```diff
@@ -68,6 +68,9 @@ public DummyQRegion(Region region) {
 
   @Override
   public boolean equals(Object o) { // for findbugs
+    if (!(o instanceof DummyQRegion)) {
```

Comment:
Not sure if we want this check here... the super implementation just checks values.  So a DummyQRegion could in theory be checked against a base QRegion and still return true.. in this case it would return false.

Reply:
My bad again. Modified the superclass to do an instance check.